File: app/src/main/java/com/spartan/karanbir/gallery/GalleryApp.java
package com.spartan.karanbir.gallery;

import android.app.Application;
import android.content.Context;

import com.danikula.videocache.HttpProxyCacheServer;

/**
 * @author karanbir
 * @since 5/23/17
 */

public class GalleryApp extends Application {
    private HttpProxyCacheServer proxy;

    public static HttpProxyCacheServer getProxy(Context context) {

        GalleryApp app = (GalleryApp) context.getApplicationContext();
        return app.proxy == null ? (app.proxy = app.newProxy()) : app.proxy;
    }

    private HttpProxyCacheServer newProxy() {
        return new HttpProxyCacheServer.Builder(this)
                .maxCacheFilesCount(6)
                .build();
    }
}

File: app/src/main/java/com/spartan/karanbir/gallery/GalleryApp.kt
package com.spartan.karanbir.gallery

import android.app.Application
import android.content.Context

import com.danikula.videocache.HttpProxyCacheServer

/**
 * @author karanbir
 * *
 * @since 5/23/17
 */

class GalleryApp : Application() {
    private var proxy: HttpProxyCacheServer? = null

    private fun newProxy(): HttpProxyCacheServer {
        return HttpProxyCacheServer.Builder(this)
                .maxCacheFilesCount(6)
                .build()
    }

    companion object {

        fun getProxy(context: Context) : HttpProxyCacheServer{
            val app = context.applicationContext as GalleryApp
            return app.proxy ?: app.newProxy()

        }
    }
}
